lib/internal/mime.js

```diff
@@ -0,0 +1,339 @@
+      StringPrototypeSearch(parameterValue, NotHTTPQuotedStringCodePoint) === -1 &&
+      paramsMap.has(parameterString) === false) {
+      paramsMap.set(parameterString, parameterValue);
+    }
+    position++;
+  }
+  return paramsMap;
+}
+
+const execPattern = (pattern, value) => {
+  pattern.lastIndex = 0;
+  return RegExpPrototypeExec(pattern, value);
+};
+
+const QUOTE_OR_SOLIDUS = hardenRegExp(/["\\]/g);
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  NotHTTPTokenCodePoint.lastIndex = 0;
+  const encode = StringPrototypeSearch(value, NotHTTPTokenCodePoint) !== -1;
+  if (!encode) return value;
+  const escaped = StringPrototypeReplace(value, QUOTE_OR_SOLIDUS, '\\$&');
+  return `"${escaped}"`;
+};
+
+const MIMEStringify = (type, subtype, parameters) => {
+  let ret = `${type}/${subtype}`;
+  const paramStr = MIMEParamsStringify(parameters);
+  if (paramStr.length) ret += `;${paramStr}`;
+  return ret;
+};
+
+const MIMEParamsStringify = (parameters) => {
+  let ret = '';
+  const entries = MIMEParamsData(parameters).entries();
+  let keyValuePair, done;
+  // Using this to avoid prototype pollution on Map iterators
+  while ({ value: keyValuePair, done } =
+        MapIteratorPrototypeNext(entries)) {
+    if (done) break;
+    const [key, value] = keyValuePair;
+    const encoded = encode(value);
+    // Ensure they are separated
+    if (ret.length) ret += ';';
+    ret += `${key}=${encoded}`;
+  }
+  return ret;
+};
+
+class MIMEParams {
+  #data;
+  constructor(init = '') {
+    this.#data = parseParametersString(`${init}`);
+  }
+
+  delete(name) {
+    this.#data.delete(name);
+  }
+
+  get(name) {
+    const data = this.#data;
+    if (data.has(name)) {
+      return data.get(name);
+    }
+    return null;
+  }
+
+  has(name) {
+    return this.#data.has(name);
+  }
+
+  set(name, value) {
+    const data = this.#data;
+    NotHTTPTokenCodePoint.lastIndex = 0;
+    name = `${name}`;
+    value = `${value}`;
+    const invalidName = StringPrototypeSearch(name, NotHTTPTokenCodePoint) !== -1;
+    if (name.length === 0 || invalidName) {
+      throw new Error('Invalid MIME parameter name');
+    }
+    NotHTTPQuotedStringCodePoint.lastIndex = 0;
+    const invalidValue = StringPrototypeSearch(value, NotHTTPQuotedStringCodePoint) !== -1;
+    if (invalidValue) {
+      throw new Error('Invalid MIME parameter value');
+    }
+    data.set(name, value);
+  }
+
+  *entries() {
+    return yield* this.#data.entries();
+  }
+
+  *keys() {
+    return yield* this.#data.keys();
+  }
+
+  *values() {
+    return yield* this.#data.values();
+  }
+
+  *[SymbolIterator]() {
+    return yield* this.#data.entries();
+  }
+
+  toJSON() {
+    return MIMEParamsStringify(this);
+  }
+
+  toString() {
+    return MIMEParamsStringify(this);
+  }
+
+  // Used to act as a friendly class to stringifying stuff
+  // not meant to be exposed to users, could inject invalid values
+  static _data(o) {
```

Comment:
Maybe have this as a `Symbol('data')` exposed from `internal/mime` so that it won't be exposed to users?

Reply:
Symbols are public via `Object.getOwnPropertySymbols`. wontfix